backend/src/main/java/hanglog/trip/dto/request/validator/RatingValidator.java

```diff
@@ -0,0 +1,17 @@
+package hanglog.trip.dto.request.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class RatingValidator implements ConstraintValidator<Rating, Double> {
+
+    @Override
+    public boolean isValid(final Double value, final ConstraintValidatorContext context) {
+        final double decimal = getDecimal(value);
+        return decimal == 0.0 || decimal == 0.5;
+    }
+
+    private double getDecimal(final Double value) {
+        return value - value.intValue();
+    }
```

Comment:
```suggestion
// suggestion 1
@Override
    public boolean isValid(final Double value, final ConstraintValidatorContext context) {
        final double decimal = value - value.intValue();
        return decimal == 0.0 || decimal == 0.5;
    }

// suggestion 2
   @Override
   public boolean isValid(final Double value, final ConstraintValidatorContext context) {
        return value >= 0 && value <= 5 && value % 0.5 == 0;
    }
```

음 굳이 getDecimal로 메서드를 나눌 필요까진 없을것 같구요.

더 나아가서 Rating 이라는 어노테이션을 따로 만들었으니
0~5 사이의 값인지도 여기서 검증해버리면 어떨까 싶기도 합니다..!
에러 메시지는 `올바르지 않은 평점 값입니다.` 정도로 추상화 시키구요..! (이건 그냥 의견임)